packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -158,66 +160,56 @@ export function dispatchEvent(
   // to filter them out until we fix the logic to handle them correctly.
   const allowReplay = (eventSystemFlags & IS_CAPTURE_PHASE) === 0;
 
-  if (
-    allowReplay &&
-    hasQueuedDiscreteEvents() &&
-    isReplayableDiscreteEvent(domEventName)
-  ) {
-    // If we already have a queue of discrete events, and this is another discrete
-    // event, then we can't dispatch it regardless of its target, since they
-    // need to dispatch in order.
-    queueDiscreteEvent(
-      null, // Flags that we're not actually blocked on anything as far as we know.
-      domEventName,
-      eventSystemFlags,
-      targetContainer,
-      nativeEvent,
-    );
-    return;
-  }
-
-  const blockedOn = attemptToDispatchEvent(
+  let blockedOn = attemptToDispatchEvent(
     domEventName,
     eventSystemFlags,
     targetContainer,
     nativeEvent,
   );
 
   if (blockedOn === null) {
-    // We successfully dispatched this event.
     if (allowReplay) {
+      // We successfully dispatched this event.
       clearIfContinuousEvent(domEventName, nativeEvent);
     }
     return;
   }
 
-  if (allowReplay) {
-    if (isReplayableDiscreteEvent(domEventName)) {
-      // This this to be replayed later once the target is available.
-      queueDiscreteEvent(
-        blockedOn,
+  if (
+    allowReplay &&
+    queueIfContinuousEvent(
+      blockedOn,
+      domEventName,
+      eventSystemFlags,
+      targetContainer,
+      nativeEvent,
+    )
+  ) {
+    return;
+  }
```

Comment:
May just be me, but I often miss the nested function calls in conditionals like this, especially if they rely on being short-circuited. What do you think about:

```suggestion
const replayWasQueued = allowReplay &&
    queueIfContinuousEvent(
      blockedOn,
      domEventName,
      eventSystemFlags,
      targetContainer,
      nativeEvent,
    );
  if (replayWasQueued) {
    return;
  }
```

Reply:
Sure I was just trying to preserve the initial style though which is why I had it like that